### PR TITLE
feat(weight-room): Today View with activity rings + quick log (#80)

### LINKED ISSUE #80: feat: strength tracker — Today View with activity rings and quick log
## Summary
Build the primary interaction surface for the strength tracker: activity rings showing daily progress and a quick-log UI for 2-tap set logging.

## Prerequisites
- lucasklawrence/courtfolio#79 (data model, storage, settings) must be implemented first

### Activity Rings (centerpiece)
- Circular SVG progress rings, one per exercise
- Outer ring: pushups (accent color, e.g. orange)
- Inner ring: pullups (secondary color, e.g. teal)
- Rings fill as sets are logged, with smooth CSS/SVG animation
- Completion state: ring glows or pulses when daily goal is hit
- Center text: total reps / goal (e.g. "75 / 100")

### Quick Log
- Tap an exercise → enter rep count → submit
- Most common action should require **minimal friction** (2 taps max)
- Default rep count suggestion (e.g. last logged amount)
- Large touch targets — mobile-first

### Today's Sets
- Timestamped list of sets logged today
- Shows running total vs. goal per exercise
- Swipe-to-delete or tap-to-remove for corrections

### Streak Counter
- Consecutive days hitting goal, displayed with a flame icon
- Resets on a missed day
- One streak per exercise

## Visual Design
- Dark theme consistent with cardio dashboard aesthetic
- Mobile-first layout — single column, large tap targets
- Activity rings as the visual hero element
- DM Mono for numbers, Instrument Serif for headings

## Implementation Details
- Create `src/strength/TodayView.tsx` as the main page component
- Create `src/strength/ActivityRings.tsx` — SVG-based circular progress
- Create `src/strength/QuickLog.tsx` — the set logging form
- Create `src/strength/SetList.tsx` — today's logged sets
- Create `src/strength/StreakBadge.tsx` — flame + count display
- Add route to the strength tracker page (React Router or conditional rendering)
- Streak computation: `lib/strengthStreaks.ts`

## Test Plan
- Unit test streak computation
- Unit test ring percentage calculations
- Verify quick-log flow works end-to-end
- Screenshot the Today View (empty state, mid-progress, goal hit) with Playwright

## Files to Create
- `src/strength/TodayView.tsx`
- `src/strength/ActivityRings.tsx`
- `src/strength/QuickLog.tsx`
- `src/strength/SetList.tsx`
- `src/strength/StreakBadge.tsx`
- `src/strength/strengthStreaks.ts`
- `src/strength/strengthStreaks.test.ts`

### PR DESCRIPTION
**Pull request by lucasklawrence**
## Summary

Builds the primary Weight Room interaction surface (PRD §7.6 / #80 of
the strength-tracker slice). Concentric SVG activity rings act as the
hero (one ring per configured exercise; outer = pushups, inner =
pullups), the quick-log surface gives admins one-tap chip presets +
"Custom" for arbitrary rep counts, and the timestamped set list owns
the per-row delete affordance. A `?preview=demo` empty-state fixture
mirrors the Combine + cardio convention so portfolio reviewers see the
rings populated without an admin account.

The route lands at `/training-facility/weight-room` (the index — slice
#82 will wire the Training Facility door + cross-view nav).

## Architecture decisions

- **Route placement: index, not sub-route.** Today View is what the
  Weight Room defaults to. History (#81) becomes `/history`; Settings
  is already at `/settings`.
- **Quick-log: preset chips + Custom.** `+5 / +10 / +15 / +20 / +25`
  one-tap chips per exercise, with a Custom button that opens an
  inline numeric input seeded by last-logged reps. Hits the issue's
  2-tap target for any count.
- **Admin-gated writes.** Same pattern as the Combine entry form —
  non-admin viewers see rings + read-only set list. Writes funnel
  through the existing `/api/admin/weight-room/sets` POST + `[id]` DELETE.
- **Demo fixture is `now`-relative.** `buildWeightRoomDemoData(new Date())`
  generates today-relative timestamps so today's ring renders mid-fill
  (the most informative ring state) and the streak badges have data.
  Generated client-side at hydration so SSR/CSR clock drift never
  lands on the user.

## What landed

| Layer | File | Notes |
| --- | --- | --- |
| Page | `app/training-facility/weight-room/page.tsx` | Server component for the flag gate + chrome; embeds the data island in a Suspense boundary. |
| Data island | `components/training-facility/weight-room/TodayDataIsland.tsx` | Owns the fetch + write orchestration. Mirrors `CombineDataIsland`'s request-id pattern so a slow mount-time fetch can't clobber post-write data. |
| Centerpiece | `components/training-facility/weight-room/ActivityRings.tsx` | SVG concentric rings, stroke-dashoffset for fill, drop-shadow glow on goal hit. |
| Quick log | `components/training-facility/weight-room/QuickLog.tsx` | Preset chips + Custom input. 44px min touch targets, ARIA-labeled buttons. |
| Set list | `components/training-facility/weight-room/SetList.tsx` | Most-recent-first, per-row delete (admin only), inline error surface. |
| Streak | `components/training-facility/weight-room/StreakBadge.tsx` | Per-exercise flame + count chip. |
| Helpers | `lib/training-facility/strength-{today,streaks}.ts` | Pure date/percent/streak math, both with unit-test coverage. |
| Demo | `constants/weight-room-demo-fixture.ts` | `buildWeightRoomDemoData(now)` — relative timestamps so today's ring populates. |
| E2E | `e2e/training-facility-{enabled,disabled}.spec.ts` | Added `/training-facility/weight-room` to the gated-route list and a render test for the Today header. |

### Drive-by fix

`lib/schemas/weight-room.ts` — rewrote `z.number().int().positive()` to a
`positiveInt()` `.refine` helper. Turbopack's dev-mode bundler
mis-transpiles the chained `.int()` when the schema gets pulled into a
client bundle (the new data island hits this), emitting a bare `int`
identifier that crashes module evaluation at runtime. Same runtime
semantics; works under Turbopack, Webpack, and the production build.

## Test plan

- [ ] CI green (Vercel + e2e + CodeRabbit).
- [ ] **Empty-state CTA.** Visit `/training-facility/weight-room` with no goals/sets — should show "No strength work logged yet" + Preview-with-sample-data button.
- [ ] **Demo fixture.** Click into `?preview=demo` → rings + streaks populate, today's pushups partially filled (75/100), pullups partially filled (20/30), Preview badge visible.
- [ ] **Exit preview.** Click "Exit preview" in the badge → returns to empty-state CTA.
- [ ] **Apply migration locally** (still required from #180 — `supabase db push`), seed at least one goal in Settings.
- [ ] **Quick-log chip.** Sign in as admin → tap `+10` on pushups → ring sweeps to ~10%, the row appears in Today's set list, custom input not opened.
- [ ] **Custom log.** Tap Custom → input seeded with last-logged value → type 17 → Log → row appears with reps=17.
- [ ] **Goal hit.** Log enough reps to cross 100% → ring locks visually at full sweep, center text shows over-fill (e.g. "120 / 100"), drop-shadow glow visible.
- [ ] **Delete a set.** Tap Delete on any row → native confirm → row disappears, ring sweeps back proportionally.
- [ ] **Read-only viewer.** Sign out → quick-log section hides, delete buttons hide, rings + set list still render.
- [ ] **Mobile viewport (`390×844`).** Rings + cards stack single-column; chips stay tappable; nothing overflows.
- [ ] **Streak grace period.** Log < goal today → today's streak should still count yesterday's hit (per `computeStrengthStreaks`).

## Out of scope

- History View / heatmap / aggregate stats — slice #81
- Weight Room door + cross-view nav on the Training Facility scene — slice #82
- Server-side daily aggregation (large-history performance) — premature; the page reads ~one day's worth of sets

Closes #80.

## Summary by CodeRabbit

* **New Features**
  * Introduced Weight Room training facility with a Today view for strength training
  * Added activity rings visualization showing daily exercise progress toward goals
  * Quick logging of exercises with preset reps or custom entry
  * History view of logged sets with timestamps and delete functionality
  * Exercise streak tracking showing current and longest consecutive days meeting goals
  * Settings link for weight room configuration

### CODE REVIEW

File: components/training-facility/weight-room/QuickLog.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  goal,
+  lastReps,
+  disabled,
+  pending,
+  onLog,
+}: QuickLogRowProps): JSX.Element {
+  const customInputId = useId()
+  const [customOpen, setCustomOpen] = useState<boolean>(false)
+  const seedCustom = lastReps ?? DEFAULT_PRESETS[1]
+  const [customValue, setCustomValue] = useState<string>(String(seedCustom))
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/QuickLog.tsx:143]: Fixed in 96d717d. Added a `useEffect` that resets `customValue` to the new seed whenever `lastReps` updates, gated on `!customOpen` so an in-progress edit is not stomped. Tested with a re-render that flips `lastReps.pushups` from 10 to 25 while the row stays mounted; the Custom input shows 25 on the next open.

File: components/training-facility/weight-room/ActivityRings.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          .map(
+            (r) =>
+              `${r.goal.exercise} ${r.totalReps} of ${r.goal.daily_target}`,
+          )
+          .join(', ')}`}
+        className="block w-full h-auto -rotate-90"
+      >
+        {rings.map((ring, i) => {
+          const radius = outerRadius - i * (RING_STROKE + RING_GAP)
+          if (radius <= RING_STROKE) return null
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/ActivityRings.tsx]: Fixed in 96d717d. Replaced the fixed `RING_STROKE` / `RING_GAP` constants with a `fitRingDimensions(size, ringCount)` helper. It picks the generous 14/4 sizes for 1–5 exercises (no visual change for the seeded `pushups` + `pullups` case) and shrinks smoothly down to 4/1 minimums for 10+ rings. The map now uses `Math.max(stroke / 2, ...)` to clamp the inner radius rather than bailing to `null`, so even a settings UI with 15+ exercises renders every configured ring. Added an 8-ring smoke test that asserts every `data-testid="ring-..."` exists.

File: components/training-facility/weight-room/QuickLog.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      <ul className="space-y-3">
+        {goals.map((goal) => (
+          <QuickLogRow
+            key={goal.exercise}
+            goal={goal}
+            lastReps={lastReps[goal.exercise]}
+            disabled={busy || (pendingExercise !== null && pendingExercise !== goal.exercise)}
+            pending={pendingExercise === goal.exercise}
+            onLog={handleLog}
+          />
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/QuickLog.tsx]: Fixed in 28c6b53. Simplified to `disabled={busy || pendingExercise !== null}` so the pending row itself locks too. Added a test that fires +10 and asserts +5 on the same row stays disabled while the POST is in flight.

File: components/training-facility/weight-room/QuickLog.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+          aria-expanded={customOpen}
+          aria-controls={customInputId}
+          className="min-h-[44px] rounded-full border border-white/15 bg-black/30 px-4 font-mono text-sm font-semibold text-white/80 transition hover:bg-white/10 disabled:cursor-not-allowed disabled:opacity-40"
+        >
+          Custom
+        </button>
+      </div>
+      {customOpen ? (
+        <form
+          id={customInputId}
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/QuickLog.tsx]: Fixed in 28c6b53. Took option A — the `<form id={customInputId}>` is now always mounted, with `hidden={!customOpen}` toggling visibility. `aria-controls` always resolves to a real DOM node. Added a test that asserts `document.getElementById(aria-controls value)` is not null.

File: components/training-facility/weight-room/SetList.tsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                  onClick={() => void handleDelete(s)}
+                  className="rounded-full border border-rose-300/25 bg-rose-300/5 px-3 py-1 font-mono text-[10px] uppercase tracking-[0.22em] text-rose-200 transition hover:bg-rose-300/15 disabled:cursor-not-allowed disabled:opacity-40"
+                >
+                  Delete
+                </button>
+              ) : null}
+            </li>
+          )
+        })}
+      </ul>
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/SetList.tsx:190]: Fixed in 28c6b53. Added a `[data-testid=set-list-totals]` strip above the row list. Each entry shows the exercise color dot, label, summed reps, and `/ daily_target` (denominator omitted when no goal exists for that exercise — covers the cascade-deleted-goal edge case my doc comment mentioned). Coverage: a totals-readout test and a no-goal denominator-omitted test.

File: components/training-facility/weight-room/StreakBadge.test.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        exercise="pushups"
+        streak={{ current: 5, longest: 12 }}
+        accentColor="#EA580C"
+      />,
+    )
+    expect(screen.getByText('pushups')).toBeInTheDocument()
+    expect(screen.getByText('5')).toBeInTheDocument()
+    expect(screen.getByText('days')).toBeInTheDocument()
+    expect(screen.getByText('12d')).toBeInTheDocument()
+  })
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/StreakBadge.test.tsx:19]: Skipping. The test passes locally (and on CI). RTL `getByText` walks the DOM and matches a normalized text node; the literal `5` is its own text node inside the parent span (the `<span>days</span>` sibling sits beside it as a separate child), so `getByText("5")` finds it. Confirmed by running just this file: `npm test -- --run StreakBadge` is green.

File: components/training-facility/weight-room/TodayDataIsland.tsx [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+
+  useEffect(() => {
+    const id = ++requestIdRef.current
+    getWeightRoomData()
+      .then((d) => {
+        if (id === requestIdRef.current) setData(d)
+      })
+      .catch(() => {
+        if (id === requestIdRef.current) setData(null)
+      })
```
[the remark this answers is not in the document]
Comment (lucasklawrence) [reply components/training-facility/weight-room/TodayDataIsland.tsx:89]: Fixed in 28c6b53. Two changes: (1) introduced a `loadError` state distinct from `data === null`, surfaced as a banner above whichever branch is rendered. (2) refetch failures now preserve the previously loaded payload (no `setData` on catch) — the user keeps their populated view and just sees the inline error. Mount-time failure still falls into the empty/CTA branch (no prior data to keep), but the banner makes the underlying state ("we couldn’t read") legible. Also tightened `realIsEmpty` to drop the `goals.length === 0` AND — goals are configuration not progress, and the migration seeds two goals so the prior predicate misclassified every fresh DB as "not empty," making `?preview=demo` a no-op. Caught while screenshotting the preview.

**Comment by lucasklawrence** [comment]
## Ready to merge

✅ **CI** — Vercel green, e2e green, CodeRabbit ✓ no blockers
✅ **Review** — `/review` self-pass, Codex (2) and CodeRabbit (4) findings all addressed in commits 96d717d + 28c6b53; one CodeRabbit flag was a false positive (RTL `getByText('5')` works because text nodes split around the inner `<span>days</span>`)

### Screenshots — Vercel preview

> All from the rebuilt preview after commit 28c6b53. Source: `?preview=demo` for the populated state, no param for the empty-state CTA.

#### `/training-facility/weight-room?preview=demo`

desktop — Today View, demo populated
mobile — Today View, demo populated

#### `/training-facility/weight-room` (empty state)

desktop — empty-state CTA
mobile — empty-state CTA

### Test plan

Click through these on the preview before merging:

- [ ] **Empty state.** Hit `/training-facility/weight-room` (no param) — should see "No strength work logged yet" + "Preview with sample data" CTA. Empty rings should NOT appear (the realIsEmpty fix).
- [ ] **Demo preview.** Click the CTA → URL gains `?preview=demo` → rings populate at 75% pushups + ~67% pullups, "Preview — sample data" badge with Exit affordance, today's set list shows 5 timestamped entries, totals row reads `pushups 75 / 100 · pullups 20 / 30`, streak badges show 8d / 5d.
- [ ] **Exit preview.** Click Exit preview in the badge → URL strips `?preview=demo` → empty-state CTA returns.
- [ ] **Mobile (`390×844`).** Same flow on mobile viewport — single-column stack, chips ≥44px tap targets, ring sized to viewport, totals row wraps cleanly.
- [ ] _(after applying the Supabase migration locally)_ Sign in as admin → Quick log surface appears → tap `+10` on pushups → ring sweeps, set row appears, set-list totals updates. Custom button opens a numeric input pre-seeded with the last preset; submitting closes the form.
- [ ] **Same-row lock.** Tap two presets back-to-back on the same exercise — only the first should fire (button stays disabled while the POST is in flight).
- [ ] **Delete.** Tap Delete on any set row → native confirm → row gone, ring sweeps back, totals decrement.
- [ ] **Read-only viewer.** Sign out → Quick log section + Delete buttons disappear; rings + set list still render.

### Merge command

```sh
gh pr merge 183 --squash --delete-branch
```

(Cleanup: the `screenshots-pr-183` branch holding the embedded images can be deleted any time after merge — `git push origin :screenshots-pr-183`.)
